### PR TITLE
UI Bottom Navigation Bar

### PR DESCRIPTION
**Pull request by AbdullahAhmedSoomro**
- Bottom Navigation Bar Completed.

### CODE REVIEW

**Comment by AbdullahAhmedSoomro** [comment]
Color will be implemented from the theme after completing this issue (Issue #298 )

File: lib/presentation/screen/main_screen.dart [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      body: Stack(
+        children: [
+          PageView(
+            controller: pageController,
+            onPageChanged: onPageChanged,
+            children: pages,
+          ),
+          CustomBottomNavBar(
+            backgroundColor: Colors.white,
+            selectedItemColor: const Color(0xFF7B7CFF),
```
Comment (hashirshoaeb) [inline lib/presentation/screen/main_screen.dart:47]: define color in theme

File: lib/presentation/screen/main_screen.dart [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  void onNavBarItemTapped(int page) {
+    pageController.jumpToPage(page);
+  }
+
+  List<Widget> pages = <Widget>[];
+
+  @override
+  Widget build(BuildContext context) {
+    return GradientScaffold(
+      body: Stack(
```
Comment (hashirshoaeb) [inline lib/presentation/screen/main_screen.dart:38]: why stack? scaffold has a property, just expose it from gradient scaffold. 
https://api.flutter.dev/flutter/material/Scaffold/bottomNavigationBar.html

File: lib/presentation/screen/main_screen.dart (hunk 1 above)
Comment (AbdullahAhmedSoomro) [reply lib/presentation/screen/main_screen.dart:47]: I commented on this above.

Color will be implemented from the theme after completing this issue #298

File: lib/presentation/shared/custom_bottom_nav_bar.dart [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }) : super(key: key);
+
+  @override
+  State<CustomBottomNavBar> createState() => _CustomBottomNavBarState();
+}
+
+class _CustomBottomNavBarState extends State<CustomBottomNavBar> {
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
```
Comment (hashirshoaeb) [inline lib/presentation/shared/custom_bottom_nav_bar.dart:31]: there is not need for stack most probably.

**Review by hashirshoaeb — CHANGES_REQUESTED** [review]
use material icons instead of images. ask designer for assistance.

**Comment by AbdullahAhmedSoomro** [comment]
Material Icons are different
image

image

On Figma
image

**Comment by hashirshoaeb** [comment]
there is no need of using Stack i guess

**Comment by hashirshoaeb** [comment]
> Material Icons are different image
> 
> image
> 
> On Figma image

use material

**Comment by AbdullahAhmedSoomro** [comment]
> there is no need of using Stack i guess

Okay, I will try this again and discuss about it

**Comment by AbdullahAhmedSoomro** [comment]
> > Material Icons are different image
> > image
> > On Figma image
> 
> use material

Okay

**Comment by AbdullahAhmedSoomro** [comment]
We can do this by using SliderTransition, which is an animation widget but we didn't animate it we give the same positions at start and end, and the duration will be 0.
But this is not the right way to do this

**Comment by AbdullahAhmedSoomro** [comment]
Filhal Merge Krdun Stack ki logic se, iska bhi aik new issue create krdunga like refactoring logic?

**Pull request merged**